server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -110,6 +114,12 @@ public AsyncQueryForwardingServlet(
     this.httpClientConfig = httpClientConfig;
     this.emitter = emitter;
     this.requestLogger = requestLogger;
+
+    // separate client with more aggressive connection timeouts
+    // to prevent cancellations requests from blocking queries
+    this.broadcastClient = httpClientProvider.get();
+    broadcastClient.setConnectTimeout(CANCELLATION_TIMEOUT);
+    broadcastClient.setMaxRequestsQueuedPerDestination(MAX_QUEUED_CANCELLATIONS);
```

Comment:
we could if we are worried about using up connection slots on the broker side. I was mostly worried about having the cancellation queue drain fast.